src/connection.ts

```diff
@@ -361,9 +361,15 @@ export class Connection {
 
   private registerDelivers() {
     this.decoder.on("deliver", async (response: DeliverResponse) => {
-      this.logger.debug(`on deliver -> ${inspect(response)} - consumers: ${this.consumers.size}`)
-      await this.askForCredit({ credit: 1, subscriptionId: response.subscriptionId })
-      response.messages.map((x) => this.consumers.get(response.subscriptionId)?.handle(x))
+      const consumer = this.consumers.get(response.subscriptionId)
+      if (consumer) {
+        this.logger.debug(`on deliver -> ${consumer.consumerId}`)
+        this.logger.debug(`response.messages.length: ${response.messages.length}`)
+        await this.askForCredit({ credit: 1, subscriptionId: response.subscriptionId })
+        response.messages.map((x) => consumer.handle(x, this.logger))
+      } else {
+        this.logger.error(`On deliver no consumer found`)
+      }
```

Comment:
```suggestion
      const consumer = this.consumers.get(response.subscriptionId)
      if (!consumer) {
        this.logger.error(`On deliver no consumer found`)
        return
      }
        this.logger.debug(`on deliver -> ${consumer.consumerId}`)
        this.logger.debug(`response.messages.length: ${response.messages.length}`)
        await this.askForCredit({ credit: 1, subscriptionId: response.subscriptionId })
        response.messages.map((x) => consumer.handle(x, this.logger))
```

https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html